README.md

```diff
@@ -37,9 +37,56 @@ Usage
 
 ZDNS was originally built as a CLI tool only. Work has been done to convert this into a library with a CLI that calls this library. Currently, the library has been separated out and a new, separate CLI has been added. Work is ongoing to clean up the interface between the CLI (or any other client program of the ZDNS library) and the ZDNS library itself.
 
-The ZDNS library lives in `github.com/zmap/zdns/pkg/zdns`. A function there, `zdns.Run()`, is used to start the ZDNS tool and do the requested lookups. Currently, this tool is intended to accept a `zdns.GlobalConf` object, `plfag` flags, and other information, but this interface is undergoing revisions to be more generally usable and continue to decouple the CLI from the library.
+Library Usage
+---------------
+
+The ZDNS library now lives in `github.com/zmap/zdns/pkg/zdns`. A function there, `zdns.Run()`, is used to start the ZDNS tool and do the requested lookups. The library is configured using a `zdns.ZdnsRun` struct. This struct contains all of the information needed to run the requested lookups. The `zdns.ZdnsRun` object contains three major groups of options:
+
+- `zdns.GlobalConf`: The Global Configuration struct holds the information that each of the Goroutines that are doing the lookup will need to perform the lookups. This includes the majority of the flags, such as timeout information, nameservers, and which module to use for the specific lookups.
+- `zdns.ModuleFlags`: The Module Flags struct holds the information needed by any of the lookup modules.
+- Miscellaneous fields: The `zdns.ZdnsRun` struct also holds configuration options that are only used for setting up the initial run, such as configuration files, local interfaces and local addresses, among others.
+
+To use the ZDNS library, a `zdns.ZdnsRun` struct must be created and passed to the `zdns.Run()` method, and then the program will read inputs from STDIN. Below is a brief example:
+
+```go
+package main
+
+import (
+	_ "github.com/zmap/zdns/pkg/alookup"
+	"github.com/zmap/zdns/pkg/zdns"
+)
+
+func main() {
+	var run zdns.ZdnsRun
+
+	// Configure this run to use the alookup module.
+	run.GlobalConf.Module = "ALOOKUP"
+
+	// Set options for all goroutines
+	run.Class = "INET"
+
+	// Run the lookup
+	zdns.Run(run)
+}
```

Comment:
I don't think this makes sense. We shouldn't have a library that blindly takes things in from STDIN.

Reply:
That's a fair point I hadn't considered, I can adjust this. We can discuss the rest of those points at our meeting so I have a clear picture. Thanks!